### PR TITLE
Rules on usage of var

### PR DESCRIPTION
**Pull request by stevemchugh**
Please review

### CODE REVIEW

File: coding/csharp.md [hunk 1]
```diff
@@ -172,6 +172,28 @@ OpenFreePopup(title, description);
...
 - Boolean variables should generally start with the prefix is, has, does or can as per methods that return boolean values
 
+#### Usage of var
+
+As a general rule of thumb usage of var should be as follows:
+
+If the right hand side tells you what the type is, the left hand side can be var
+
+```csharp
+var report = newReport(x, x, x);
```
Comment (westonized) [inline coding/csharp.md]: space missing? or method name with bad capitalization?

**Comment by westonized** [comment]
Edge cases for clarification:
```
var var1 = "This is clearly a string.";
var var2 = 27;
var var3 = Convert.ToInt32(Console.ReadLine());
```

**Comment by stevemchugh** [comment]
Your edge cases added

**Comment by westonized** [comment]
@stevemchugh why is it not a problem that we don't know the type of `Console.ReadLine()`?

**Comment by stevemchugh** [comment]
ReSharper suggests it should be 

string line = Console.ReadLine();

File: coding/csharp.md (hunk 1 above)
Comment (stevemchugh) [inline coding/csharp.md]: Can you elaborate?

Comment (westonized) [inline coding/csharp.md]: Is `newReport` a method? or just missing a space i.e. `new Report` (I know from context it's the latter)

**Pull request merged**